mozilla_schema_generator/glean_ping.py

```diff
@@ -57,6 +57,30 @@ def __init__(self, repo, **kwargs):  # TODO: Make env-url optional
             **kwargs,
         )
 
+    def get_schema(self, generic_schema=False) -> Schema:
+        """
+        Fetch schema via URL.
+
+        Unless *generic_schema* is set to true, this function makes some modifications
+        to allow some workarounds for proper injection of metrics.
+        """
+        schema = super().get_schema()
+        if generic_schema:
+            return schema
+
+        # We need to inject placeholders for the url2, text2, etc. types as part
+        # of mitigation for https://bugzilla.mozilla.org/show_bug.cgi?id=1737656
+        for metric_name in ["labeled_rate", "jwe", "url", "text"]:
+            metric1 = schema.get(
+                ("properties", "metrics", "properties", metric_name)
+            ).copy()
+            metric1 = schema.set_schema_elem(
+                ("properties", "metrics", "properties", metric_name + "2"),
+                metric1,
+            )
```

Comment:
I thought we were only going to add the 2 suffix for existing pings, but this definitely looks like it will do it for all pings?

Reply:
This adds in a placeholder to the glean json schema for all pings. When metrics get injected later on, these fields disappear if no metrics are defined.

But it is true that this change as written means that all pings that define a `url`-type metric will have that metric put in place under `url2`. Downstream logic will be more complex if it has to handle both the `url2` case and the `url` case with correct type. 

But I definitely see the argument that it's desirable for new pings and apps to have the fully "correct" structure. It's mostly just view generation that will be affected.

Reply:
Much as it pains me to say this, I'm leaning towards thinking it's probably better if we just use the `2` suffix for all the things. Realistically we're probably going to be stuck with this for a while and it's going to be easier to deal with if the behaviour is consistent.

However, this comment might do with some slight updating to make sure the current behaviour is clear:

https://github.com/mozilla/mozilla-schema-generator/blob/b21d2ad0374ccf58e7c071da946d8c9a76e0a9e5/mozilla_schema_generator/glean_ping.py#L191

I guess we're assuming that no one has actually landed any of these metric types since #213 was merged?

Reply:
We are assuming that no one has landed metrics of any of these 4 types over all time. And it's important that we get this merged before an app does try to implement any of these.

Reply:
I've updated the commentary in `glean_ping.py` to make the new "2" behavior clear.